cmd/ci-secret-bootstrap/main.go

```diff
@@ -0,0 +1,182 @@
+	if err != nil {
+		return err
+	}
+	o.kubeConfigs = kubeConfigs
+
+	config, err := loadConfig(o.configPath)
+	if err != nil {
+		return err
+	}
+	o.config = config
+
+	return validateConfig(kubeConfigs, config)
+}
+
+type Config struct {
+	Secrets []SecretConfig `json:"secrets"`
+}
+
+type BWContext struct {
+	BWItem     string `json:"bwItem"`
+	Field      string `json:"field"`
+	Attachment string `json:"attachment"`
+}
+
+type SecretContext struct {
+	Cluster   string `json:"cluster"`
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+}
+
+type SecretConfig struct {
+	From map[string]BWContext `json:"from"`
+	To   []SecretContext      `json:"to"`
+}
+
+func loadConfig(path string) (*Config, error) {
+	bytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	c := &Config{}
+	err = yaml.Unmarshal(bytes, c)
+	if err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+func validateConfig(kConfigs map[string]rest.Config, c *Config) error {
+	//TODO
+	// case1: either field or attachment but not both
+	// case2: cluster is a context in the kubeconfig
+	return nil
+}
+
+func main() {
+	o := parseOptions()
+	if err := o.validateOptions(); err != nil {
+		logrus.WithError(err).Fatal("Invalid arguments.")
+	}
+	if err := o.completeOptions(); err != nil {
+		logrus.WithError(err).Fatal("Failed to complete options.")
+	}
+	bwClient := bitwarden.NewBitwardenClient()
+	if err := bwClient.Login(o.bwUser, o.bwPassword); err != nil {
+		logrus.WithError(err).Fatal("Failed to login Bitwarden.")
+	}
+	defer func() {
+		if err := bwClient.Logout(); err != nil {
+			logrus.WithError(err).Fatal("Failed to login Bitwarden.")
+		}
+	}()
+
+	for _, secretConfig := range o.config.Secrets {
+		data := make(map[string][]byte)
+		for key, bwContext := range secretConfig.From {
+			if bwContext.Field != "" {
+				fieldValue, err := bwClient.GetField(bwContext.BWItem, bwContext.Field)
+				if err != nil {
+					logrus.WithError(err).Fatalf("Failed to get field value '%s' in item '%s'", bwContext.Field, bwContext.BWItem)
+				}
+				data[key] = []byte(fieldValue)
+			} else {
+				attachmentValue, err := bwClient.GetAttachment(bwContext.BWItem, bwContext.Attachment, true)
+				if err != nil {
+					logrus.WithError(err).Fatalf("Failed to get attachment value '%s' in item '%s'", bwContext.Attachment, bwContext.BWItem)
+				}
+				data[key] = []byte(attachmentValue)
+			}
+		}
+		for _, secretContext := range secretConfig.To {
+			secret := &coreapi.Secret{
+				Data: data,
+				ObjectMeta: meta.ObjectMeta{
+					Name: secretContext.Name,
+				},
+			}
+			kc := o.kubeConfigs[secretContext.Cluster]
+			client, err := coreclientset.NewForConfig(&kc)
+			if err != nil {
+				logrus.WithError(err).Fatalf("could not get core client for cluster '%s'", secretContext.Cluster)
+			}
+			_, err = client.Secrets(secretContext.Namespace).Create(secret)
+			if err != nil {
+				if kerrors.IsAlreadyExists(err) {
```

Comment:
Inverting the condition and turning it into a guard clause would allow you to reduce some indentation depth

```
if !kerrors.IsAlreadyExists(err) {
  error out
}
if _, err := client.Secrets(secretContext.Namespace).Update(secret); err != nil {
  error out
}
```

Reply:
Smart!
Updated.

One comment: Removing the else-part (for reduction of code complication) replies on the Fatal/Panic to keep the logic right.